### PR TITLE
DM-41978: Switch logic used to exclude tasks

### PR DESCRIPTION
**Pull request by leeskelvin**

### CODE REVIEW

File: python/lsst/source/injection/utils/make_injection_pipeline.py [hunk 1]
```diff
@@ -155,27 +155,17 @@ def make_injection_pipeline(
...
-            for host_subset in host_subsets:
-                pipeline.removeLabelFromSubset(host_subset, task_label)
-        # Then remove the task from the pipeline.
-        try:
-            pipeline.removeTask(task_label)
-        except KeyError:
-            not_excluded_tasks.add(task_label)
-    if len(not_excluded_tasks) > 0:
-        grammar = "Task" if len(not_excluded_tasks) == 1 else "Tasks"
+    all_tasks = {taskDef.label for taskDef in pipeline.toExpandedPipeline()}
```
Comment (leeskelvin) [inline python/lsst/source/injection/utils/make_injection_pipeline.py:158]: Does this need to be `toExpandedPipeline`? (A question for @natelust)

Comment (leeskelvin) [reply python/lsst/source/injection/utils/make_injection_pipeline.py:158]: For context: here we're just looping over all tasks in the pipeline to get a set of task names. No actual task/config manipulation is performed in this block.

Comment (leeskelvin) [reply python/lsst/source/injection/utils/make_injection_pipeline.py:158]: It's never quite clear for me when I need to expand my pipeline, and when I can get away with using the base pipeline object. I.e., I want to be sure that I'm including all the tasks from imported pipeline YAMLs as well.

**Review by jtmccann — APPROVED** [review]
Looks good to me!

**Pull request merged**